### ISSUE TITLE
TypeError caused by credentials.js

### ISSUE THREAD

**Issue by jacekkopecky** [grpc/grpc#7171]
Using `gcloud` and giving it a nonexistent key file leads to a type error:

```
/XXX/node_modules/grpc/src/node/src/credentials.js:108
      callback(code, message, metadata._getCoreRepresentation());
      ^

TypeError: The callback's first argument must be a status code
    at TypeError (native)
    at /XXX/node_modules/grpc/src/node/src/credentials.js:108:7
    at /XXX/node_modules/grpc/src/node/src/credentials.js:124:9
    at /XXX/node_modules/google-auth-library/lib/auth/oauth2client.js:286:14
    at /XXX/node_modules/google-auth-library/lib/auth/jwtclient.js:138:16
    at ReadFileContext.handleJSONKey [as callback] (/XXX/node_modules/gtoken/lib/index.js:83:7)
    at FSReqWrap.readFileAfterOpen [as oncomplete] (fs.js:303:13)
```

This seems to be caused by [these lines in `credentials.js`](https://github.com/grpc/grpc/blob/9b9708a9c484c02ad8ef27060370ae605a83942c/src/node/src/credentials.js#L101-L105) because the FS error comes with an `ENOENT` `code` but the native grpc binding expects a grpc status code.

### Expected behaviour:

my code gets a callback with an error

### Observed behaviour:

my code never gets a callback, the console prints out the above message

### Workaround:

Commenting out lines 101-103 and 105 in `credentials.js` makes my code receive the right error callback.

**Comment by felipesabino** [comment]
I am experiencing the same issue but not with a non-exiting key. I get it using `gcloud.pubsub` and setting  `credentials` with  `client_email` and `private_key` after just a couple of seconds sending data.

```
Auth error: { [Error: socket hang up] code: 'ECONNRESET' }
XXX/node_modules/gcloud/node_modules/grpc/src/node/src/credentials.js:108
      callback(code, message, metadata._getCoreRepresentation());
      ^
TypeError: The callback's first argument must be a status code
  at TypeError (native)
  at XXX/node_modules/gcloud/node_modules/grpc/src/node/src/credentials.js:108:7
  at XXX/node_modules/gcloud/node_modules/grpc/src/node/src/credentials.js:124:9
  at XXX/node_modules/gcloud/node_modules/google-auto-auth/node_modules/google-auth-library/l…
  at XXX/node_modules/gcloud/node_modules/google-auto-auth/node_modules/google-auth-library/l…
  at Request._callback (XXX/node_modules/gcloud/node_modules/google-auto-auth/node_modules/go…
  at self.callback (XXX/node_modules/gcloud/node_modules/google-auto-auth/node_modules/google…
  at Request.emit (events.js:107:17)
  at Request.onRequestError (XXX/node_modules/gcloud/node_modules/google-auto-auth/node_modul…
  at ClientRequest.emit (events.js:107:17)
  at TLSSocket.socketOnEnd (_http_client.js:293:9)
  at TLSSocket.emit (events.js:129:20)
  at _stream_readable.js:908:16
  at process._tickCallback (node.js:355:11)
```

**Comment by murgatroid99** [comment]
This should now be fixed in gRPC version `1.0.0-pre2`
